### PR TITLE
Fix 449

### PR DESCRIPTION
**Pull request by jonesh66**
When Selenium 4 is being used, 
1. Change path to `/status`.
2. Append `standalone` to seleniumArgs

### CODE REVIEW

File: lib/default-config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = () => {
   const config = {
     baseURL: 'https://selenium-release.storage.googleapis.com',
-    version: '3.141.59',
+    version: '4.0.0-alpha-3',
```
Comment (mgrybyk) [inline lib/default-config.js:4]: Please revert this change

**Comment by jonesh66** [comment]
@mgrybyk will add tests for this change as soon as I figure out where to add them

File: lib/get-selenium-status-url.js [hunk 2]
```diff
@@ -68,7 +69,7 @@ exports.getSeleniumStatusUrl = function (seleniumArgs) {
   }
 
   const statusURI = new URL('http://' + host);
-  const nodeStatusAPIPath = '/wd/hub/status';
+  const nodeStatusAPIPath = defaultConfig.version.startsWith('4.') ? '/status' : '/wd/hub/status';
```
Comment (mgrybyk) [inline lib/get-selenium-status-url.js:72]: Don't use default config to determine version

File: lib/start.js [hunk 3]
```diff
@@ -32,6 +32,10 @@ function start(_opts, _cb) {
     opts.spawnCb = noop;
   }
 
+  if (defaultConfig.version.startsWith('4.')) {
```
Comment (mgrybyk) [inline lib/start.js:35]: Don't use default config for version detection

**Review by mgrybyk — CHANGES_REQUESTED** [review]

File: lib/start.js [hunk 4]
```diff
@@ -32,6 +32,10 @@ function start(_opts, _cb) {
...
+    opts.seleniumArgs.push('standalone');
```
Comment (mgrybyk) [inline lib/start.js:36]: Only push it if it's not already there

File: lib/start.js (hunk 3 above)
Comment (jonesh66) [reply lib/start.js]: using opts now

File: lib/get-selenium-status-url.js (hunk 2 above)
Comment (jonesh66) [reply lib/get-selenium-status-url.js]: using opts here

File: lib/default-config.js (hunk 1 above)
Comment (mgrybyk) [reply lib/default-config.js:4]: This one still should be reverted

File: lib/get-selenium-status-url.js [hunk 5]
```diff
@@ -35,6 +35,8 @@ exports.getSeleniumStatusUrl = function (seleniumArgs) {
   const nodeConfigArg = seleniumArgs.indexOf('-nodeConfig');
   const portArg = seleniumArgs.indexOf('-port');
   const hostArg = seleniumArgs.indexOf('-host');
+  const isSelenium4 = seleniumArgs[4];
```
Comment (mgrybyk) [inline lib/get-selenium-status-url.js:38]: Please don't rely on index

**Review by mgrybyk — CHANGES_REQUESTED** [review]

File: lib/get-selenium-status-url.js (hunk 5 above)
Comment (jonesh66) [reply lib/get-selenium-status-url.js:38]: what argument could be used to determine the version? 
I want to use something like `const isSelenium4 = seleniumArgs.indexOf('-version')`

File: lib/start.js [hunk 6]
```diff
@@ -32,6 +32,10 @@ function start(_opts, _cb) {
     opts.spawnCb = noop;
   }
 
+  if (opts.version.startsWith('4.') && opts.seleniumArgs.indexOf('standalone') === -1) {
```
Comment (mgrybyk) [inline lib/start.js:35]: I forgot to take into account that we should only add standalone if the mode is not specified explicitly.
https://www.automatetheplanet.com/selenium-webdriver-4-overview/
```
// add somewhere to the top
const seleniumModes = ['standalone', 'hub', 'node']

// replace opts.seleniumArgs.indexOf('standalone') === -1 with
!seleniumModes.some((a) => seleniumArgs.includes(a))
```

File: lib/get-selenium-status-url.js (hunk 5 above)
Comment (mgrybyk) [reply lib/get-selenium-status-url.js:38]: let's pass `opts` to `checkStarted` function, then pass it to `getSeleniumStatusUrl`.

Don't use args to detect version, use opts instead

**Review by mgrybyk — CHANGES_REQUESTED** [review]

File: lib/start.js (hunk 6 above)
Comment (mgrybyk) [reply lib/start.js:35]: Hm, I think it would be even better if at least one arg passed and that's it. 
```suggestion
  if (opts.version.startsWith('4.') && opts.seleniumArgs.length === 0) {
```

File: lib/get-selenium-status-url.js (hunk 5 above)
Comment (jonesh66) [reply lib/get-selenium-status-url.js:38]: I haven't been able to figure out a way to pass opts to checkstarted.
`async function checkStarted(seleniumArgs, opts) {...`
how do I know what value does opts hold here? I think it has something to do with `checkArgs` but I am not sure how to proceed. Could you please advise?

Comment (mgrybyk) [reply lib/get-selenium-status-url.js:38]: you can use debugger and follow the call chain to figure out which function is calling checkStarted function

image

`checkPathsExistence` -> `checkStarted`

so, you have to pass opts to checkPathsExistence, then pass opts from checkPathsExistence to checkStarted

Comment (mgrybyk) [reply lib/get-selenium-status-url.js:38]: following that I'm able to pass opts
image

Comment (mgrybyk) [reply lib/get-selenium-status-url.js:38]: here you go https://github.com/vvo/selenium-standalone/compare/pass-opts-to-check-started

**Comment by mgrybyk** [comment]
@jonesh66 do you need any help?

**Comment by jonesh66** [comment]
@mgrybyk I think I'm using opts now but the tests are failing

**Comment by mgrybyk** [comment]
@jonesh66 see logs, looks like you are trying to get version from undefined.
Please fix tests

**Comment by mgrybyk** [comment]
Can I help with something else here?

File: lib/default-config.js (hunk 1 above)
Comment (mgrybyk) [reply lib/default-config.js:4]: Ping

Comment (jonesh66) [reply lib/default-config.js:4]: sorry I was still testing this using the config. I'll revert this to the original once the tests pass. (i know it's messy)
